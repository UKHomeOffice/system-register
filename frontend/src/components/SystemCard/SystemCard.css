div.systemCard {
    padding: 6px;
    margin-left: -6px;
    margin-top: 6px;
  }

.systemCard > a{
    display:inline-block;
    text-decoration: none;
    width:100%;
    height:100%;
  }

  .systemCard-description {
    margin-top: 3px;
  }

 .systemCard a:focus, .systemCard a:active {
    background-color: var(--yellow);
 }