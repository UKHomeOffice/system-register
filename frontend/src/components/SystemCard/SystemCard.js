import React from 'react'
import './SystemCard.css'
import Link from '../Linking/Link'
import ModifiedDetails from '../ModifiedDetails/ModifiedDetails'

const trimDescription = (str) => {
    if (!str) {
        return "No Description"
    }
    if (str.length > 200) {
        return str.substring(0, 200) + '...'
    }
    return str
}

const SystemCard = (props) => { //todo code review, extract to reusable styles "LINK" component
    const { id, name, portfolio, description, last_updated } = props.system
    return (
        <div className="systemCard">
            <Link style={{color: "black"}} className="systemCard-link" to={`/system/${id}`}>
                <strong>{name}</strong>
                <small> (Portfolio: {portfolio}, <ModifiedDetails date={last_updated.timestamp} author_name={last_updated.author_name} />)
                </small>
                <div>
                    <small>{trimDescription(description)}</small>
                </div>
            </Link>
        </div>
    )
}

export default SystemCard
