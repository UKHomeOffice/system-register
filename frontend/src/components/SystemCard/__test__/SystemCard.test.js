import React from 'react'
import { render, getByLabelText } from '@testing-library/react'
import SystemCard from '../SystemCard'
import renderer from 'react-test-renderer'
import { BrowserRouter } from 'react-router-dom'

const up_to_date_system = {
    name: "System X",
    portfolio: 'Portfolio Y',
    last_updated: '2020-02-13T11:01:01'
}

const never_updated_system = {
    name: "System X",
    portfolio: 'Portfolio Y',
    last_updated: ''
}

describe('<SystemCard />', () => {
    it('renders name', () => {
        const { getByText } = render(
            <BrowserRouter>
                <SystemCard system={up_to_date_system} />
            </BrowserRouter>)
        const element = getByText(up_to_date_system.name)
        expect(element).toBeInTheDocument()
    });

    it('renders portfolio', () => {
        const { getByText } = render(
            <BrowserRouter>
                <SystemCard system={up_to_date_system} />
            </BrowserRouter>)
        const element = getByText(/Portfolio Y/)
        expect(element).toBeInTheDocument()
    });

    it('renders last_updated', () => {
        const { getByText } = render(
            <BrowserRouter>
                <SystemCard system={up_to_date_system} />
            </BrowserRouter>)
        const element = getByText(/February 2020/)
        expect(element).toBeInTheDocument()
    });

    it('renders last_updated as Never when falsy', () => {
        const { getByText } = render(
            <BrowserRouter>
                <SystemCard system={never_updated_system} />
            </BrowserRouter>)
        const element = getByText(/Never/) //todo need to be more specific in case
        expect(element).toBeInTheDocument()
    });

    it('matches snapshot', () => {
        const snapshot = renderer.create(<BrowserRouter><SystemCard system={up_to_date_system} /></BrowserRouter>)
        expect(snapshot).toMatchSnapshot()
    });
})