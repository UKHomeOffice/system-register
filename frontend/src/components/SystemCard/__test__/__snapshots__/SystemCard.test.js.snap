// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SystemCard /> matches snapshot 1`] = `
<div
  className="systemCard topMargin"
>
  <a
    href="/system/undefined"
    onClick={[Function]}
  >
    <strong>
      System X
    </strong>
    <small>
       (Portfolio: 
      Portfolio Y
      , 
      <span
        data-testid="modified-on"
      >
        Last modified: 
        13 February 2020
      </span>
      )
    </small>
    <div
      style={
        Object {
          "marginTop": "3px",
        }
      }
    >
      <small>
        No Description
      </small>
    </div>
  </a>
</div>
`;
